File: internal/interfaces/services/info/network.go
// SPDX-License-Identifier: BSD-2-Clause
//
// Copyright (c) 2025 The FreeBSD Foundation.
//
// This software was developed by Hayzam Sherif <hayzam@alchemilla.io>
// of Alchemilla Ventures Pvt. Ltd. <hello@alchemilla.io>,
// under sponsorship from the FreeBSD Foundation.

package infoServiceInterfaces

import "time"

type NetworkInterface struct {
	Name    string `json:"name"`
	Flags   string `json:"flags"`
	Network string `json:"network"`
	Address string `json:"address"`

	ReceivedPackets int64 `json:"received-packets"`
	ReceivedErrors  int64 `json:"received-errors"`
	DroppedPackets  int64 `json:"dropped-packets"`
	ReceivedBytes   int64 `json:"received-bytes"`

	SentPackets int64 `json:"sent-packets"`
	SendErrors  int64 `json:"send-errors"`
	SentBytes   int64 `json:"sent-bytes"`

	Collisions int64 `json:"collisions"`
}

type HistoricalNetworkInterface struct {
	SentBytes     int64     `json:"sentBytes"`
	ReceivedBytes int64     `json:"receivedBytes"`
	CreatedAt     time.Time `json:"createdAt"`
}

File: new-web/src/lib/types/info/network.ts
import { z } from 'zod/v4';

export const NetworkInterfaceInfoSchema = z.object({
	id: z.number().default(0),
	name: z.string().default(''),
	flags: z.string().default(''),
	network: z.string().default(''),
	address: z.string().default(''),
	receivedPackets: z.number().int().default(0),
	receivedErrors: z.number().int().default(0),
	droppedPackets: z.number().int().default(0),
	receivedBytes: z.number().int().default(0),
	sentPackets: z.number().int().default(0),
	sendErrors: z.number().int().default(0),
	sentBytes: z.number().int().default(0),
	collisions: z.number().int().default(0),
	createdAt: z.string(),
	updatedAt: z.string()
});

export const HistoricalNetworkInterfaceSchema = z.object({
	sentBytes: z.number().int().default(0),
	receivedBytes: z.number().int().default(0),
	createdAt: z.string()
});

export type HistoricalNetworkInterface = z.infer<typeof HistoricalNetworkInterfaceSchema>;
export type NetworkInterfaceInfo = z.infer<typeof NetworkInterfaceInfoSchema>;
